python/tchannel/tornado/tchannel.py

```diff
@@ -159,13 +172,57 @@ def handle_stream(self, stream, address):
         })
 
         log.debug(
-            "Successfully completed handshake with %s (%s)",
+            "Successfully completed handshake with %s:%s (%s)",
             conn.remote_host,
+            conn.remote_host_port,
             conn.remote_process_name,
         )
 
-        self.tchannel.peers.get(conn.remote_host).register_incoming(conn)
+        self.tchannel.peers.get(
+            "%s:%s" % (conn.remote_host,
+                       conn.remote_host_port)
```

Comment:
When the handshake is received, the remote_host is set to whatever was received in the handshake -- which is the address at which the peer can be reached. You don't need to manually concatenate this. This will probably result in `$ipAddress:$port:$port`.